openedx/core/djangoapps/discussions/docs/decisions/0006-discussions-api-fetch-inactive-providers.rst

```diff
@@ -0,0 +1,82 @@
+Allow fetching configuration for inactive providers
+===================================================
+
+
+Status
+------
+
+Proposal
+
+
+Context
+-------
+
+In ADR 003 an API was proposed for discussion configuration. This proposed a
+single API that returned information about the active provide and listed other
+available providers.
+
+The API was designed to only return information about the current active
+provider however in practise this creates an issue when switching providers.
+
+The troublesome scenario is as follows:
+
+- A course author configures the legacy provider
+- They switch to another provider
+- Now they switch back to the legacy provider
+- They have to set up the legacy provider from scratch again
+
+Since we can't fetch configuration for an inactive provider, the wizard cannot
+use their existing configuration for the legacy provider.
+
+Requirements
+------------
+
+A way to fetch provider data for a provider without needing to switch to it
+first.
+
+Alternatives Considered
+-----------------------
+
+An alternative that was considered was simply updating the current provider
+each time configuration for a different provider is needed, however this is
+very disruptive for what should be a simple task.
+
+Decision
+--------
+
+We can split the existing configuration API into two parts. One that simply
+lists all the providers. This API is purely read-only.
+
+The other API is a read-write API the configuration data for the current
+provider or any other provider as it takes the ``provider_id`` as an argument.
+
+The current API will be split into the following two endpoints:
+
+- ``/api/discussions/v0/course/{course_id}/providers``:
+
+  This will return read-only provider config data. This will return the
+  following pieces of information:
+
+    - active: the active provider
+    - features: feature support levels
+    - available: provider listing
+
+- ``/api/discussions/v0/course/{course_id}/settings``:
+
+  This is the read-write settings API that can take an optional `provider_id`
+  argument in which case returns data for that provider.
+
+  This will return the following information which is common for all providers:
+
+    - enabled
+    - provider_type
+    - enable_in_context
+    - enable_graded_units
+    - unit_level_visibility
+
+  The following information will vary based on provider specified:
+
+    - lti_configuration: only returned for the active provider.
+    - plugin_configuration: plugin configuration for the specified provider.
+
+  Changing settings for an inactive provider is out of scope here.
```

Comment:
can you copy these data args into an expected data structure? so that is more clear. I mean 
```
{
    ...
    enabled: True,
    provider_type: 'piazza'
    ...
}
```

Reply:
Also, its unclear, how this data will change with/without providing a specific _discussion provider_.

Reply:
It is following the same API as right now, so instead of listing the data structure I've only listed the elements that will change or be retained. Will update.